File: leetcode/1957-delete-characters-to-make-fancy-string/1957-delete-characters-to-make-fancy-string.java
class Solution {
    public String makeFancyString(String s) {
        int n = s.length();
        char prev = s.charAt(0);
        int freq = 1;
        StringBuffer ans = new StringBuffer();
        ans.append(prev);

        for (int i =1; i<n; i++) {
            if (s.charAt(i) == prev) freq++;
            else {
                prev = s.charAt(i);
                freq = 1;
            }
            if (freq < 3) ans.append(prev);
        }
        return ans.toString();
    }
}

File: leetcode/1957-delete-characters-to-make-fancy-string/1957-delete-characters-to-make-fancy-string.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        ans = ""
        n = len(s)
        cur = 0
        while cur < n:
            leng = 0
            ch = s[cur]
            while cur < n and s[cur] == ch:
                leng += 1
                cur +=1
            leng = 2 if leng > 2 else leng
            ans += ch * leng
        return ans